File: backend/components/metiers/threed/tests_centralises/templates/helpers/validators/test_templates_validator.js
// test_templates_validator.js – Tests avancés helpers/validators (JS)
const path = require('path');
const fs = require('fs');
const { isValidTemplateFile, validateTemplateContent } = require('../../../../templates/helpers/validators/templates_validator');

describe('Templates Helpers Validators Ultra Advanced', () => {
    it('should validate template file existence and extension', () => {
        const filePath = path.join(__dirname, 'mock_template.html.j2');
        fs.writeFileSync(filePath, '<div>{{ var }}</div>', 'utf8');
        expect(isValidTemplateFile(filePath)).toBe(true);
        fs.unlinkSync(filePath);
        expect(isValidTemplateFile(filePath)).toBe(false);
        expect(isValidTemplateFile('invalid.exe')).toBe(false);
    });

    it('should validate template content (Jinja2, HTML, JSON, edge cases)', () => {
        expect(validateTemplateContent('<div>{{ var }}</div>')).toBe(true);
        expect(validateTemplateContent('{ "a": 1 }')).toBe(true);
        expect(validateTemplateContent('no tags')).toBe(false);
        expect(validateTemplateContent('')).toBe(false);
    });

    it('should handle edge cases and CI/CD compliance', () => {
        expect(validateTemplateContent('{{}}')).toBe(true);
        expect(validateTemplateContent('<>')).toBe(true);
        expect(validateTemplateContent('{}')).toBe(true);
    });
});

File: backend/components/metiers/threed/tests_centralises/templates/helpers/validators/test_templates_validator.py
import pytest
from threed.templates.helpers.validators.templates_validator import is_valid_template_file, validate_template_content
import tempfile
import os

def test_is_valid_template_file(tmp_path):
    # Fichier valide
    valid_file = tmp_path / "template.html.j2"
    valid_file.write_text("<div>{{ var }}</div>")
    assert is_valid_template_file(str(valid_file))
    # Fichier inexistant
    assert not is_valid_template_file(str(tmp_path / "notfound.html.j2"))
    # Extension invalide
    invalid_file = tmp_path / "invalid.exe"
    invalid_file.write_text("")
    assert not is_valid_template_file(str(invalid_file))

def test_validate_template_content():
    assert validate_template_content('<div>{{ var }}</div>')
    assert validate_template_content('{ "a": 1 }')
    assert not validate_template_content('no tags')
    assert not validate_template_content('')
    # Edge cases
    assert validate_template_content('{{}}')
    assert validate_template_content('<>')
    assert validate_template_content('{}')
